From d1dc51f95f9e831aa6adbbeff549c2017d2dfd82 Mon Sep 17 00:00:00 2001
From: Andrew Udvare <audvare@gmail.com>
Date: Tue, 29 Dec 2020 15:18:06 -0500
Subject: [PATCH] configure.ac: allow --enable-python=system on macOS if PYTHON is non-empty

This will work as long as a valid Python is in PATH, such as /usr/bin/python3
from Xcode or another version in some prefix like /opt/local.

Change-Id: Ic967c3ce2f9949d94c11c3449363841a1565cfa9
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/108486
Tested-by: Jenkins
Reviewed-by: Michael Stahl <michael.stahl@allotropia.de>
---

diff --git configure.ac configure.ac
index e660733..78bb799 100644
--- configure.ac
+++ configure.ac
@@ -9238,8 +9238,9 @@
 # Optionally user can pass an option to configure, i. e.
 # ./configure PYTHON=/usr/bin/python
 # =====================================================================
-if test $_os = Darwin -a "$enable_python" != no -a "$enable_python" != fully-internal -a "$enable_python" != internal; then
+if test $_os = Darwin -a "$enable_python" != no -a "$enable_python" != fully-internal -a "$enable_python" != internal -a "$enable_python" != system; then
     # Only allowed choices for macOS are 'no', 'internal' (default), and 'fully-internal'
+    # unless PYTHON is defined as above which allows 'system'
     enable_python=internal
 fi
 if test "$build_os" != "cygwin" -a "$enable_python" != fully-internal; then
@@ -9311,7 +9312,7 @@
     ;;
 system)
     AC_MSG_RESULT([system])
-    if test "$_os" = Darwin; then
+    if test "$_os" = Darwin -a -z "$PYTHON"; then
         AC_MSG_ERROR([--enable-python=system doesn't work on macOS because the version provided is obsolete])
     fi
     ;;
